pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -29,71 +29,98 @@ import (
 	samplescheme "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned/scheme"
 	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions/etcd/v1alpha1"
 	listers "github.com/xmudrii/etcdproxy-controller/pkg/client/listers/etcd/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-const controllerAgentName = "etcdstorage-controller"
+const controllerAgentName = "etcdproxy-controller"
 
 const (
 	// SuccessSynced is used as part of the Event 'reason' when a EtcdStorage is synced
 	SuccessSynced = "Synced"
 	// ErrResourceExists is used as part of the Event 'reason' when a EtcdStorage fails
-	// to sync due to a Deployment of the same name already existing.
+	// to sync due to a ReplicaSet of the same name already existing.
 	ErrResourceExists = "ErrResourceExists"
+	// ErrUnknown is used as part of the Event 'reason' when a EtcdStorage fails
+	// to get, create, or update resource.
+	ErrUnknown = "ErrUnknown"
 
 	// MessageResourceExists is the message used for Events when a resource
-	// fails to sync due to a Deployment already existing
+	// fails to sync due to a ReplicaSet already existing
 	MessageResourceExists = "Resource %q already exists and is not managed by EtcdStorage"
 	// MessageResourceSynced is the message used for an Event fired when a EtcdStorage
 	// is synced successfully
 	MessageResourceSynced = "EtcdStorage synced successfully"
 )
 
+// EtcdConnectionInfo type is used to wire information
+// used by controller to create ReplicaSets.
+type EtcdConnectionInfo struct {
+	EtcdURL             string
+	EtcdCAConfigMapName string
+	EtcdCertSecretName  string
+}
+
 // Controller is the controller implementation for EtcdStorage resources
 type Controller struct {
 	// kubeclientset is a standard kubernetes clientset
 	kubeclientset kubernetes.Interface
-	// sampleclientset is a clientset for our own API group
-	sampleclientset clientset.Interface
+	// etcdProxyClient is a clientset for our own API group
+	etcdProxyClient clientset.Interface
+
+	replicasetsLister appslisters.ReplicaSetLister
+	replicasetsSynced cache.InformerSynced
+
+	servicesLister corev1listers.ServiceLister
+	servicesSynced cache.InformerSynced
 
-	deploymentsLister  appslisters.DeploymentLister
-	deploymentsSynced  cache.InformerSynced
 	etcdstoragesLister listers.EtcdStorageLister
 	etcdstoragesSynced cache.InformerSynced
 
-	// workqueue is a rate limited work queue. This is used to queue work to be
-	// processed instead of performing it as soon as a change happens. This
-	// means we can ensure we only process a fixed amount of resources at a
-	// time, and makes it easy to ensure we are never processing the same item
-	// simultaneously in two different workers.
 	workqueue workqueue.RateLimitingInterface
 	// recorder is an event recorder for recording Event resources to the
 	// Kubernetes API.
 	recorder record.EventRecorder
+
+	// etcdConnectionInfo is used to wire information
```

Comment:
optional nit: I take my lines to 120 or so.

Reply:
I have no hard rule on this, but I see my IDE is showing line on 120 as well, so let's use it.